### DISCUSSION TITLE
Flip Efect for ebooks

### DISCUSSION THREAD

**Discussion by KaanAlper** [sumatrapdfreader/sumatrapdf#3959] [Ideas]
In apps like Honeyview or etc there is a effects for page transistions,  for online readers like me that can be good to there is a flip effects 

https://github.com/sumatrapdfreader/sumatrapdf/assets/86665266/3716139d-8ce0-4757-9a06-55787df99fc1

**Comment by GitHubRulesOK** [comment]
Those flip books in general are usually JavaScript using rendered images they cannot  usually be read aloud as its not text. SumatraPDF is fast reader, safer without JavaScript to exploit, and not an online editor/converter. here we see this PDF is converted to images that are then used in a SLIDESHOW with "flipping" effect.
<img width="793" height="571" alt="image" src="https://github.com/user-attachments/assets/bc33694e-d67f-4788-8266-5247f8a2ee74" />

**Comment by AlAinCoder** [comment]
Feature Request:
Could you please introduce a **Page Flip effect** with a page corner curling and a soft page turning sound (as in online flipbooks)?
It can be hard coded into the Sumatra PDF app as an option in Bookmode (without enabling Javascript).

No Javascript.
Just an animated Page Flip effect option.

Thanks.
It will be a major plus in app features.

**Reply by GitHubRulesOK** [reply]
@AlAinCoder 
Note I am not the developer but in my humble opinion this would be restricted to only a few formats.

Images are fixed density so to create the AI (Artificial Illusion) of page turning they are temporarily converted and WARPED this is really a very inefficient use of time between reading pages that can be done best in JavaScript as designed for lightest fastest web users function.

The use of the inbuilt web viewer to cover ALL such documents AND LAYOUTS as SumatraPDF supports means some cannot easily be converted for "flipping" it simply would be a bigger programming task that has little merit to the majority.

So SumatraPDF can convert this web page into 2 images as seen on the left and Honeyview can even reverse the pixels for TRUE image flipping then show them like in an online flipbook but that all takes a lot of time to process in-between each page view.

The MAJOR point is the pages MUST be converted into STATIC images to then be presented as a slideshow with false flipping.

<img width="1089" height="683" alt="image" src="https://github.com/user-attachments/assets/5dae9435-17f6-4090-b9ce-ed7f8294c645" />

**Comment by GitHubRulesOK** [comment]
@AlAinCoder 
Using JavaScript then media e.g. flipbook inside SumatraPDF for Audio-Video or PDF & ePub is not a problem for example 

https://github.com/user-attachments/assets/d7a4b27f-69f8-482d-8fab-4dcf4cbc1464
